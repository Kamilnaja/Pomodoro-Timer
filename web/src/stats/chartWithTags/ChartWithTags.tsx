import React from 'react';
import { Line, LineChart } from 'recharts';
import { Pomodoro } from '../../../../types/statsInterfaces';
import { ChartWithTagsProps } from './ChartWithTagsProps';

const getCount = (arr: Pomodoro[], idx: number) =>
  arr.filter(item => new Date(item.createdAt).getDate() === new Date(arr[idx].createdAt).getDate()).length;

export interface PomodorosInDays {
  day: {
    count: number;
    pomodoro: Pomodoro[];
  };
}

export const ChartWithTags = (props: ChartWithTagsProps) => {
  const mappedProps = props.statsWithTags.result.map((item: Pomodoro, idx: number, arr: Pomodoro[]): {
    month: number;
    day: number;
    count: number;
    tagText: string;
    createdAt: string;
  } => ({
    ...item,
    month: new Date(item.createdAt).getMonth(),
    day: new Date(item.createdAt).getDate(),
    count: getCount(arr, idx),
  }));

  return (
    <div>
      <LineChart width={400} height={400} data={mappedProps}>
        <Line type="monotone" dataKey="count" stroke="#8884d8" />
      </LineChart>
    </div>
  );
};
