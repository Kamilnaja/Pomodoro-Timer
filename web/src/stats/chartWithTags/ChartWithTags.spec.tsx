import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { createPomodoroSearchResult } from '../testing/stats.test.data';
import { ChartWithTags } from './ChartWithTags';

describe('ChartWithTags', () => {
  let wrapper: ShallowWrapper;
  const statsWithTags = createPomodoroSearchResult();

  beforeEach(() => {
    wrapper = shallow(<ChartWithTags statsWithTags={statsWithTags} />);
  });

  it('should match snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
