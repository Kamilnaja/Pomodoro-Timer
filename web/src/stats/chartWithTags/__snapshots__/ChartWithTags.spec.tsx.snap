// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartWithTags should match snapshot 1`] = `
<div>
  <LineChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "count": 3,
          "createdAt": "2021-03-01T05:47:01.476Z",
          "day": 1,
          "month": 2,
          "tagText": null,
        },
        Object {
          "count": 3,
          "createdAt": "2021-03-01T06:13:36.785Z",
          "day": 1,
          "month": 2,
          "tagText": null,
        },
        Object {
          "count": 3,
          "createdAt": "2021-03-01T17:28:16.182Z",
          "day": 1,
          "month": 2,
          "tagText": null,
        },
        Object {
          "count": 1,
          "createdAt": "2021-03-02T19:59:00.403Z",
          "day": 2,
          "month": 2,
          "tagText": null,
        },
      ]
    }
    height={400}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
    syncMethod="index"
    width={400}
  >
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="count"
      dot={true}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stroke="#8884d8"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
  </LineChart>
</div>
`;
